crates/metrics/src/lib.rs

```diff
@@ -17,19 +26,27 @@ const DEFAULT_SERVICE_NAME: &str = "unknown_service";
 
 static SERVICE_NAME: Mutex<Option<&str>> = Mutex::new(None);
 
+pub type MeterProviderBuilderIntercept = fn(MeterProviderBuilder) -> MeterProviderBuilder;
+static METER_PROVIDER_BUILDER: Mutex<Option<MeterProviderBuilderIntercept>> = Mutex::new(None);
+
 static METRICS_CORE: Lazy<Arc<ServiceMetrics>> = Lazy::new(|| {
     let service_name = SERVICE_NAME.lock().unwrap().unwrap_or(DEFAULT_SERVICE_NAME);
+    let meter_provider_builder = *METER_PROVIDER_BUILDER.lock().unwrap();
 
     let registry = Registry::new();
     let prometheus_exporter = opentelemetry_prometheus::exporter()
         .with_registry(registry.clone())
         .build()
         .unwrap();
-    let provider = SdkMeterProvider::builder()
-        .with_reader(prometheus_exporter)
-        .build();
+    let mut builder = SdkMeterProvider::builder().with_reader(prometheus_exporter);
+    if let Some(with_view) = meter_provider_builder {
```

Comment:
I'd rename `with_view` -> `builder_fn`.

Reply:
Renamed in [bd56a39](https://github.com/WalletConnect/utils-rs/pull/18/commits/bd56a395154758f1082d304b671daba926eae595)